File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        # brute force 
        # put it into a set, sort the set
        # iterate it through once 

        count = sorted(set(nums))
        longest = 0
        curr = 1

        for i in range(1, len(count)):
            if count[i - 1] + 1 == count[i]:
                curr += 1
            else:
                longest = max(curr, longest)
                curr = 1

        return max(longest, curr)

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums, target) {
        // brute force 
        let start = 0;
        let end = nums.length - 1;

        while (start < end) {
            if (nums[start] === target) {
                return start;
            } else if (nums[end] === target) {
                return end;
            }
            start++;
            end--; 
        }

        return -1;
    }
}
